drivers/sdhc/sdhc_renesas_ra.c

```diff
@@ -0,0 +1,718 @@
+		p_csd_reg.reg.sdrsp10 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP10 << 8;
+		p_csd_reg.reg.sdrsp32 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP32 << 8;
+		p_csd_reg.reg.sdrsp54 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP54 << 8;
+		p_csd_reg.reg.sdrsp76 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP76 << 8;
+
+		memcpy(cmd->response, &p_csd_reg.reg, sizeof(cmd->response));
+	} else {
+		/* Fill response buffer */
+		p_csd_reg.reg.sdrsp10 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP10;
+		p_csd_reg.reg.sdrsp32 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP32;
+		p_csd_reg.reg.sdrsp54 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP54;
+		p_csd_reg.reg.sdrsp76 = (uint32_t)priv->sdmmc_ctrl.p_reg->SD_RSP76;
+
+		memcpy(cmd->response, &p_csd_reg.reg, sizeof(cmd->response));
+	}
+end:
+	if (cmd->opcode == SD_APP_CMD) {
+		priv->app_cmd = true;
+	} else {
+		priv->app_cmd = false;
+	}
+
+	return ret;
+}
+
+static int sdhc_ra_reset(const struct device *dev)
+{
+	struct sdhc_ra_priv *priv = dev->data;
+	const struct sdhc_ra_config *cfg = dev->config;
+
+	k_sem_init(&priv->thread_lock, 1, 1);
+	k_sem_take(&priv->thread_lock, K_USEC(50));
+
+	/* Reset SDHI. */
+	((R_SDHI0_Type *)cfg->regs)->SOFT_RST = 0x0U;
+	((R_SDHI0_Type *)cfg->regs)->SOFT_RST = 0x1U;
+
+	k_sem_give(&priv->thread_lock);
+
+	return 0;
+}
+
+/*
+ * Set SDHC io properties
+ */
+static int sdhc_ra_set_io(const struct device *dev, struct sdhc_io *ios)
+{
+	struct sdhc_ra_priv *priv = dev->data;
+	const struct sdhc_ra_config *cfg = dev->config;
+	struct st_sdmmc_instance_ctrl *p_ctrl = &priv->sdmmc_ctrl;
+	int fsp_err;
+	int ret = 0;
+
+	uint8_t bus_width;
+	uint32_t bus_width_reg;
+
+	if (ios->bus_width > 0) {
+		bus_width_reg = 0;
+		/* Set bus width, SD bus interface doesn't support 8BIT */
+		switch (ios->bus_width) {
+		case SDHC_BUS_WIDTH1BIT:
+			bus_width = 1;
+			bus_width_reg = 4;
+			break;
+		case SDHC_BUS_WIDTH4BIT:
+			bus_width = 4;
+			break;
+		default:
+			ret = -ENOTSUP;
+			goto end;
+		}
+
+		if (priv->bus_width != bus_width) {
+			/* Set the bus width in the SDHI peripheral. */
+			((R_SDHI0_Type *)cfg->regs)->SD_OPTION =
+				SDHI_PRV_SD_OPTION_DEFAULT |
+				(bus_width_reg << SDHI_PRV_SD_OPTION_WIDTH8_BIT);
+			priv->bus_width = bus_width;
+		}
+	}
+
+	if (ios->clock) {
+		if (ios->clock > priv->props.f_max || ios->clock < priv->props.f_min) {
+			LOG_ERR("Proposed clock outside supported host range");
+			return -EINVAL;
+		}
+
+		if (priv->bus_clock != (uint32_t)ios->clock) {
+			fsp_err = r_sdhi_max_clock_rate_set(p_ctrl, ios->clock);
+			ret = err_fsp2zep(fsp_err);
+			if (ret < 0) {
+				goto end;
+			}
+			priv->bus_clock = ios->clock;
+		}
+	}
+
+	if (ios->timing > 0) {
+		/* Set I/O timing */
+		if (priv->timing != ios->timing) {
+			switch (ios->timing) {
+			case SDHC_TIMING_LEGACY:
+			case SDHC_TIMING_HS:
+				break;
+			case SDHC_TIMING_DDR50:
+			case SDHC_TIMING_DDR52:
+				/* Enable DDR mode */
+				LOG_INF("DDR mode enabled");
```

Comment:
Not clear here- is there supposed to be some change to the hardware applied? Or is that done elsewhere?

Reply:
Sorry for the mistake here. Currently, we don't support DDR mode in our hardware so I removed it.